src/Symfony/Component/Security/Http/MagicLink/MagicLoginLinkDetails.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\MagicLink;
+
+/**
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class MagicLoginLinkDetails
+{
+    private $url;
+    private $expiresAt;
+
+    public function __construct(string $url, \DateTimeInterface $expiresAt)
```

Comment:
I would force using `\DateTimeImmutable`, to guarantee that this class is immutable (passing a `DateTime` would allow mutating the state of the link details by doing `$details->getExpiresAt()->setTimestamp(...)` for instance)